### PR TITLE
Surface error message to the user when setting the `stream` flag to true

### PR DESCRIPTION
**Pull request by cw75**
## Describe your changes and why you are making these changes
Previously, when we set the `stream` flag to true, we are only streaming the error message to the server log but not returning the error to the user. This PR make it so that the error log is also captured and returned to the user (instead of just saying "Error waiting for command to complete" which is not useful).

## Checklist before requesting a review
- [ ] I have created a descriptive PR title. The PR title should complete the sentence "This PR...".
- [ ] I have performed a self-review of my code.
- [ ] I have included a small demo of the changes. For the UI, this would be a screenshot or a Loom video.
- [ ] If this is a new feature, I have added unit tests and integration tests.
- [ ] I have run the integration tests locally and they are passing.
- [ ] I have run the linter script locally (See `python3 scripts/run_linters.py -h` for usage).
- [ ] All features on the UI continue to work correctly.
- [ ] Added one of the following CI labels:
    - `run_integration_test`: Runs integration tests
    - `skip_integration_test`: Skips integration tests (Should be used when changes are ONLY documentation/UI)

### CODE REVIEW

File: src/golang/lib/lib_utils/utils.go
Comment (kenxu95) [inline src/golang/lib/lib_utils/utils.go:100]: I don't know too much about the bufio.Scanner class, but this is guaranteed to return even if nothing is ever logged to stderr? How does it know that...

File: src/golang/lib/lib_utils/utils.go
Comment (kenxu95) [inline src/golang/lib/lib_utils/utils.go:109]: hmm I would expect this to block forever until something is written to ch... is something guaranteed to be written?

**Review by kenxu95 — COMMENTED** [review]
Hmm this channel thing looks a little fishy to me. Does this really work if nothing is ever written to ch?

File: src/golang/lib/lib_utils/utils.go
Comment (cw75) [reply src/golang/lib/lib_utils/utils.go:100]: Yes, when the program exits, the scanner will reach the end and break out of the for loop.

File: src/golang/lib/lib_utils/utils.go
Comment (cw75) [reply src/golang/lib/lib_utils/utils.go:109]: and yes you can send an empty string to channel and the listener blocked on the channel will get unblocked.

**Comment by cw75** [comment]
@kenxu95 I can write a couple comments clarifying my response if that's helpful.

**Comment by kenxu95** [comment]
@cw75 yeah that would be helpful. I was just confused because:
1) I had no idea when the Scan() would terminate
2) I did not know sending an empty string to the channel would unblock it.

**Comment by cw75** [comment]
@kenxu95 addressed comments.

**Review by kenxu95 — APPROVED** [review]

**Pull request merged**
